### PR TITLE
bishop allocation

### PR DESCRIPTION
**Pull request by driedyam**

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -218,10 +218,22 @@ function balance(pool, message){
 		groups[0] = [leaders[0], leaders[1], ...pool]
 	} else {
 		message.channel.send("Zakum has detected two groups! Balancing groups...")
-		let difference = 1000;
+		let difference = 1000; // this being used?
```
Comment (tommymurphy) [inline bot.js]: I don't think so 😅

File: bot.js [hunk 2]
```diff
@@ -218,10 +218,22 @@ function balance(pool, message){
...
+		let poolToJoin = pool.filter(person => !person.leader && person.role.toLowerCase() != 'bishop') // gimmie my non-leaders/non-bishops
+
+		/* 
+			 ALLOCATE BISHOPS
+			 assumptions:
+			 - groups may have existing members in them
+		*/
+		bishops.forEach(function(b){
+			if(!g0.map(g => g.role.toLowerCase()).includes('bishop')){ g0.push(b); }	// give g0 a bish if they don't have one
+			else if(!g1.map(g => g.role).includes('bishop')){ g1.push(b) } // give g1 a bish if they don't have one
```
Comment (tommymurphy) [inline bot.js]: Missing g.role.toLowerCase() here?

Comment (tommymurphy) [reply bot.js]: Will g0 always get the best Bishop every time with this logic?

Comment (driedyam) [reply bot.js]: Would you prefer to get the weaker bishop?

Comment (tommymurphy) [reply bot.js]: I won't always be a raid leader

Comment (tommymurphy) [reply bot.js]: I think the leader ranks need to be compared here

File: bot.js [hunk 3]
```diff
@@ -218,10 +218,20 @@ function balance(pool, message){
...
+
+		// ALLOCATE BISHOPS
+		bishops.forEach(function(b, i){
+			prioritizeGroups([g0,g1]).forEach(function(group){
+				if(group.map(g => g.role.toLowerCase()).includes('bishop')) return;
+				group.push(b);
+				bishops.splice(i, 1);
+			})
+		})
+		bishops.forEach(b => poolToJoin.push(b)) // dump unused bishops back into pool
```
Comment (tommymurphy) [inline bot.js]: This is kind of disrespectful to the unused bishops

**Pull request merged**